contrib/sentry.php

```diff
@@ -192,14 +188,49 @@ static function (&$value) use ($config) {
     }
 );
 
-function releaseIsGitDirectory()
+function getPreviousReleaseRevision()
 {
-    return (bool) run('cd {{release_path}} && git rev-parse --git-dir > /dev/null 2>&1 && echo 1 || echo 0');
+    switch (get('update_code_strategy')) {
+        case 'archive':
+            if (has('previous_release')) {
+                return run('cat {{previous_release}}/REVISION');
+            }
+
+            return null;
+        case 'clone':
+            if (has('previous_release')) {
+                cd('{{previous_release}}');
+                return trim(run('git rev-parse HEAD'));
+            }
+
+            return null;
+        default:
+            throw new ConfigurationException(parse("Unknown `update_code_strategy` option: {{update_code_strategy}}."));
+    }
+}
+
+function getCurrentReleaseRevision()
+{
+    switch (get('update_code_strategy')) {
+        case 'archive':
+            return run('cat {{release_path}}/REVISION');
+
+        case 'clone':
+            cd('{{release_path}}');
+            return trim(run('git rev-parse HEAD'));
+
+        default:
+            throw new ConfigurationException(parse("Unknown `update_code_strategy` option: {{update_code_strategy}}."));
+    }
 }
 
 function getReleaseGitRef(): Closure
 {
     return static function ($config = []): string {
+        if (get('update_code_strategy') === 'archive') {
+            return run('cat {{current_path}}/REVISION');
```

Comment:
Same here, if a custom `git_version_command` has been provided, you'd better `cd('{{deploy_path}}/.dep/repo');` then run the custom `git_version_command` instead.